Sprint_7

rediyss opened the pull request.

rediyss committed:
Тесты на
-Создание курьера
-Логин курьера
-Создание заказа
-Список заказов

rediyss committed:
Добавление отчета allure

okryukova commented on pom.xml:
⚠️Можно улучшить. Хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройку (если необходима) и как запускать (mvn clean test - кавычки позволяют выполнять запуск из файла README.md). Для оформления данных в виде таблицы используй сочетание клавиш Ctrl + N или Cmd + N. Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+
+    @Before
+    public void setup() {
+        RestAssured.baseURI = "https://qa-scooter.praktikum-services.ru";
```
⛔️Нужно исправить.  Base URL присваиваем один раз в проекте. Можно использовать request specification в RestAssured или утилитный базовый класс.

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+    }
+
+    @Test
+    @DisplayName("Можно создать курьера")
```
⚠️Можно улучшить. В allure есть еще одна аннотация для описания теста Description. Добавь ее, пожалуйста, во всех тестах проекта

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+
+    @Test
+    @DisplayName("Можно создать курьера")
+    public void canCreateCourier() {
```
⚠️Можно улучшить. В названии метода необходимо использовать слово Test. Проверь, соблюдается ли данное правило для всех созданных тестов.

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+    @Test
+    @DisplayName("Можно создать курьера")
+    public void canCreateCourier() {
+        Createuser courier = new Createuser("Artas12345991", "1234", "Rediys");
```
⚠️Можно улучшить. Для создания курьера используй случайные данные. Лучше использовать готовые решения JavaFaker/DataFaker, так как это сокращает нашу работу.

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+        Createuser courier = new Createuser("Artas12345991", "1234", "Rediys");
+
+        Response createResponse = createCourier(courier);
+        createResponse.then().statusCode(201).body("ok", is(true));
```
⚠️Можно улучшить. Вместо числовых кодов ответов, лучше использовать коды из библиотеки static org.apache.http.HttpStatus, так как они там строковые и несут смысл, что повышает читаемость и не надо вспоминать, что есть что. Пример кода, statusCode(SC_CREATED).

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+        createResponse.then().statusCode(201).body("ok", is(true));
+
+        // Получаем ID для удаления
+        courierId = getCourierId(courier.getLogin(), courier.getPassword());
```
⛔️Нужно исправить. Этот  шаг нужно перенести в метод tearDown() . Он должен выполняться даже если проверка в тесте не прошла. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+    @Test
+    @DisplayName("Нельзя создать двух одинаковых курьеров")
+    public void cannotCreateDuplicateCourier() {
+        Createuser courier = new Createuser("Artas12345991", "1234", "Rediyss");
```
⛔️Нужно исправить.  Создание сущности курьер можно вынести в метод с аннотацией before

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+        Response response = createCourier(courier);
+        response.then().statusCode(400).body("message", not(empty()));
+        // Не устанавливаем courierId, так как курьер не создан
+    }
```
⛔️Нужно исправить. Не проверен сценарий - создание курьера без пароля 
⚠️Можно улучшить. Поле firstName является необязательным для курьера. Попробуй создать курьера без этого поля.

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+        return given()
+                .contentType(ContentType.JSON)
+                .body(courier)
+                .post("/api/v1/courier");
```
⚠️Можно улучшить. Константы с указанием пути можно вынести в отдельный класс, так с ними будет удобнее работать. Назови его, например, Endpoints

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,100 @@
+                .when()
+                .delete("/api/v1/courier/" + id)
+                .then()
+                .statusCode(200);
```
⛔️Нужно исправить. В методах, в которых отправляются запросы, не должно быть проверок. Методов, в котором отправляется запрос и делается проверка, не является универсальным . Проверки можно делать в тестах или писать отдельные методы.

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierLoginTest.java:
```diff
@@ -0,0 +1,156 @@
+    @Test
+    @DisplayName("Для авторизации нужны все обязательные поля")
+    public void loginRequiresAllFields() {
+        // Без логина
```
⛔️Нужно исправить. Для каждого кейса нужен отдельный тест. Это две независимые проверки

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierLoginTest.java:
```diff
@@ -0,0 +1,156 @@
+    @Test
+    @DisplayName("Неверный логин или пароль возвращает ошибку")
+    public void wrongLoginOrPasswordReturnsError() {
+        // Неверный логин
```
⛔️Нужно исправить.  Аналогично, два теста , а  не один

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierLoginTest.java:
```diff
@@ -0,0 +1,156 @@
+    }
+
+    @Step("Удаление курьера")
+    private void deleteCourier(String id) {
```
⚠️Можно улучшить. Много дублирования кода по формированию и отправления запроса. Постаряйся избегать дублирование кода. Можно создать классы CourierApi и OrderApi. В каждом классе создать методы для формирования запроса (например: создание courier, удаление courier, логин и т.д.). Проверку ответа делать или в тесте, или можно написать отдельные методы. В этом случае, если изменится запрос для создания курьера, тебе придется вносить правки в одном месте (в методе), а не во все тесты сразу.

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/Createuser.java:
```diff
@@ -0,0 +1,40 @@
+package ru.praktikum_services.qa_scooter.tests;
+
+public class Createuser {
```
⛔️Нужно исправить. Неверный нейминг. Каждое новое слово с заглавной буквы

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/Createuser.java:
```diff
@@ -0,0 +1,40 @@
+package ru.praktikum_services.qa_scooter.tests;
+
+public class Createuser {
+    private String password;
```
⚠️Можно улучшить. Можно использовать @DaTa и @AllArgsConstructor из библиотеки Lombok

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/LoginCourier.java:
```diff
@@ -0,0 +1,29 @@
+package ru.praktikum_services.qa_scooter.tests;
```
⚠️Можно улучшить.  Группируем классы по пакетам: test, model, step

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/OrderCreateTest.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> testData() {
```
⚠️Можно улучшить. В параметризованных тестах для аннотации Parameterized.Parameters лучше использовать аргумент name: @Parameterized.Parameters(name = "Тестовые данные: {0} {1}"), где {0}, {1} - индексы параметров. Это повысит информативность проверки

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/OrderCreateTest.java:
```diff
@@ -0,0 +1,62 @@
+        order.put("comment", "bebeb");
+        order.put("color", colors);
+
+        Response response = given()
```
⛔️Нужно исправить. В тесте мы не используем напрямую библиотеку rest-assured, не отправляем запросы. Код для отправки запросов необходимо выделять в отдельный метод и располагать его в классе для отправки запросов. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/OrderCreateTest.java:
```diff
@@ -0,0 +1,62 @@
+    @Test
+    @DisplayName("Создание заказа с параметрами цвета")
+    public void createOrderWithColors() {
+        Map<String, Object> order = new HashMap<>();
```
⛔️Нужно исправить. Нужно использовать социализацию

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/OrderCreateTest.java:
```diff
@@ -0,0 +1,62 @@
+
+        response.then().statusCode(201).body("track", notNullValue());
+    }
+}
```
⛔️Нужно исправить. Созданный заказ нужно отменить после выполнения теста

rediyss committed:
Анотация Parameterized.Parameters и добавление комментарий к заказам для лучшего понимая какой этап

rediyss committed:
Исправлен BaseURL на requestSpec, слуайно зарезолвивл комм с этим исправлением

rediyss committed:
Исправлено разделены тесты
-Авторизация только с паролем
-Авторизация с логином

-Авторизация с неверным логином
-Авторизация с нерным паролем

rediyss committed:
Созданный заказ отменяется
Код для отправки запроссов был веделен в отдельный класс

rediyss committed:
Добалвен класс Order для конструктора заказов
Обновлен OrderClient и OrderCreateTest.java

rediyss committed:
Получение ID для курьера и удаление его вынесен в After

rediyss committed:
Убрана проверка в методе

rediyss committed:
Добавлено два теста без пароля и без firstName

rediyss committed:
Убран лишний импорт

rediyss committed:
Исправлен нейминг Createuser на CreateUser

rediyss committed:
Добавлен Lombok

rediyss committed:
Добавлен @Description в тесты и улучшен нейминг тестов

rediyss committed:
улучшен нейминг тестов

rediyss committed:
Почищен код

rediyss committed:
Добавлен отчет

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,129 @@
+    @Before
+    public void setup() {
+        requestSpec = new RequestSpecBuilder()
+                .setBaseUri("https://qa-scooter.praktikum-services.ru")
```
⛔️Нужно исправить.  Base URL присваиваем один раз в проекте. Можно использовать request specification в RestAssured или утилитный базовый класс.

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/CourierCreateTest.java:
```diff
@@ -0,0 +1,129 @@
+        courier.setFirstName("Rediyss");
+
+        Response response = createCourier(courier);
+        response.then().statusCode(400).body("message", not(empty()));
```
⛔️Нужно исправить. Нужно проверить корректность сообщения об ошибке в теле. Поправь, пожалуйста, во всем коде

okryukova commented on src/test/java/ru/praktikum_services/qa_scooter/tests/OrderCreateTest.java:
```diff
@@ -0,0 +1,67 @@
+                .extract()
+                .path("track");
+
+        orderClient.cancelOrder(track);
```
⛔️Нужно исправить. Нужно вынести этот шаг в метод с аннотацией after

okryukova commented on target/allure-results/allure.properties:
⛔️Нужно исправить. Нужно приложить всю папку allure-results

rediyss committed:
Base URL вынесен в отдельный класс

rediyss committed:
Отмена заказа вынесена в @After

rediyss committed:
Отмена заказа вынесена в @After

rediyss committed:
Добавлена проверка коректности сообщения об ошибке

rediyss committed:
Удален ненунжный импорт

rediyss committed:
Добалвен allure отчет
